src/MSBuild/XMake.cs

```diff
@@ -1206,9 +1213,17 @@ string outputResultsCache
 #endif
                     BuildResultCode? result = null;
 
-                    var messagesToLogInBuildLoggers = Traits.Instance.EscapeHatches.DoNotSendDeferredMessagesToBuildManager
-                        ? null
-                        : GetMessagesToLogInBuildLoggers();
+                    IEnumerable<BuildManager.DeferredBuildMessage> messagesToLogInBuildLoggers = null;
+                    if (!Traits.Instance.EscapeHatches.DoNotSendDeferredMessagesToBuildManager)
```

Comment:
I agree. I have refrained from inverting and renaming it because this escape hatch is optout with intuitive understanding that 'true => optout'. Give this code future maintenance needs, I'd keep it as is.
Additionally, can we consider to delete this Escape Hatch.